x-pack/test/functional/services/ml/job_table.ts

```diff
@@ -207,12 +224,16 @@ export function MachineLearningJobTableProvider(
       return !subSelector ? row : `${row} > ${subSelector}`;
     }
 
-    public async withDetailsOpen<T>(jobId: string, block: () => Promise<T>): Promise<T> {
+    public async withDetailsOpen<T>(
+      jobId: string,
+      block: () => Promise<T>,
+      ensureClosed: boolean = true
+    ): Promise<T> {
       await this.ensureDetailsOpen(jobId);
       try {
         return await block();
       } finally {
-        await this.ensureDetailsClosed(jobId);
+        if (ensureClosed) await this.ensureDetailsClosed(jobId);
```

Comment:
I think this is unnecessarily complicated. If I want just the details open and don't want to ensure they're closed again, I could just call `ensureDetailsOpen(jobId)` instead of `withDetailsOpen(jobId, block, false)`. Is there a reason for this construction that I don't see?

Reply:
I see what you mean. No, not all.